src/storage-client/src/controller.rs

```diff
@@ -856,7 +856,7 @@ where
             {
                 Some(
                     durable_metadata
-                        .remove(&status_collection_id)
+                        .get(&status_collection_id)
```

Comment:
If you're waiting on aljoscha's return to re-review: is this a bug we should merge separately? I haven't thought through the implications of `remove` instead of `get`.